experiments/generic/pod-delete/pod-delete.go

```diff
@@ -0,0 +1,146 @@
+	var GracePeriod int64 = 0
+
+	for x := 0; x < experimentsDetails.Iterations; x++ {
+		//Getting the list of all the target pod for deletion
+		targetPodList, err := PreparePodList(experimentsDetails, clients)
+		if err != nil {
+			return err
+		}
+		log.InfoWithValues("[Info]: Killing the following pods", logrus.Fields{
+			"PodList": targetPodList})
+
+		//Deleting the application pod
+		for _, pods := range targetPodList {
+			if experimentsDetails.Force == true {
+				err = clients.KubeClient.CoreV1().Pods(experimentsDetails.AppNS).Delete(pods, &v1.DeleteOptions{GracePeriodSeconds: &GracePeriod})
+			} else {
+				err = clients.KubeClient.CoreV1().Pods(experimentsDetails.AppNS).Delete(pods, &v1.DeleteOptions{})
+			}
+		}
+		if err != nil {
+			return err
+		}
+
+		if experimentsDetails.EngineName != "" {
+			msg := "Injecting " + experimentsDetails.ExperimentName + " chaos on application pod"
+			environment.SetEventAttributes(eventsDetails, types.ChaosInject, msg)
+			events.GenerateEvents(experimentsDetails, clients, eventsDetails)
+		}
+
+		//Waiting for the chaos interval after chaos injection
+		if experimentsDetails.ChaosInterval != 0 {
+			log.Infof("[Wait]: Wait for the chaos interval %vs", strconv.Itoa(experimentsDetails.ChaosInterval))
+			waitForChaosInterval(experimentsDetails)
+		}
+		//Verify the status of pod after the chaos injection
+		log.Info("[Status]: Verification for the recreation of application pod")
+		err = status.CheckApplicationStatus(experimentsDetails.AppNS, experimentsDetails.AppLabel, clients)
+
+		//ChaosCurrentTimeStamp contains the current timestamp
+		ChaosCurrentTimeStamp := time.Now().Unix()
+
+		//ChaosDiffTimeStamp contains the difference of current timestamp and start timestamp
+		//It will helpful to track the total chaos duration
+		chaosDiffTimeStamp := ChaosCurrentTimeStamp - ChaosStartTimeStamp
+
+		if int(chaosDiffTimeStamp) >= experimentsDetails.ChaosDuration {
+			break
+		}
+
+	}
+	log.Infof("[Completion]: %v chaos is done", experimentsDetails.ExperimentName)
+
+	return nil
+}
+
+//waitForChaosInterval waits for the given ramp time duration (in seconds)
+func waitForChaosInterval(experimentsDetails *types.ExperimentDetails) {
+	time.Sleep(time.Duration(experimentsDetails.ChaosInterval) * time.Second)
+}
+
+//PreparePodList derive the list of target pod for deletion
+//It is based on the KillCount value
+func PreparePodList(experimentsDetails *types.ExperimentDetails, clients environment.ClientSets) ([]string, error) {
+
+	var targetPodList []string
+
+	err := retry.
+		Times(90).
+		Wait(2 * time.Second).
+		Try(func(attempt uint) error {
+			pods, err := clients.KubeClient.CoreV1().Pods(experimentsDetails.AppNS).List(v1.ListOptions{LabelSelector: experimentsDetails.AppLabel})
+			if err != nil || len(pods.Items) == 0 {
+				return errors.Errorf("Unable to get the pod, err: %v", err)
+			}
+			//Adding the first pod only, if KillCount is not set or 0
```

Comment:
lets randomize selection in cases where: kill count is not specified (means single/1) OR where kill_count < len(pods).